### PR TITLE
Obsolete Environment.Properties

### PR DESCRIPTION
**Pull request by hazzik**
The bigger idea is to gradually move away from static configurations.

### CODE REVIEW

File: src/NHibernate/Dialect/Dialect.cs [hunk 1]
```diff
@@ -148,15 +148,18 @@ protected Dialect()
 		public static Dialect GetDialect()
```
Comment (fredericDelaporte) [inline src/NHibernate/Dialect/Dialect.cs]: Shouldn't this one be obsoleted too ? How would it work without `Environment.Properties` ?

File: src/NHibernate/Cfg/Configuration.cs [hunk 2]
```diff
@@ -171,7 +171,9 @@ protected void Reset()
 			propertyReferences = new List<Mappings.PropertyReference>();
 			FilterDefinitions = new Dictionary<string, FilterDefinition>();
 			interceptor = EmptyInterceptor.Instance;
+#pragma warning disable 618
 			properties = Environment.Properties;
```
Comment (fredericDelaporte) [inline src/NHibernate/Cfg/Configuration.cs]: For fully preparing its removal, the code here should read the configuration file. (Or eventually cache statically in `Configuration` that file read result as a dictionary.)

Comment (hazzik) [reply src/NHibernate/Cfg/Configuration.cs]: I decided to split this into several PRs. This deserves full re-implementation.

File: src/NHibernate/Dialect/Dialect.cs (hunk 1 above)
Comment (hazzik) [reply src/NHibernate/Dialect/Dialect.cs]: I decided to split this into several PRs.

**Comment by hazzik** [comment]
The reason why this is just obsoleted is because I wanted to make this PR focused as much as possible.

**Pull request merged**